lib/elixir/lib/protocol.ex

```diff
@@ -116,6 +116,73 @@ defmodule Protocol do
 
   @doc """
   Derives the `protocol` for `module` with the given options.
+
+  If your implementation passes options, you will also
+  need to provide a macro named
+  `__deriving__(module, struct, options)` to get the
+  options that were passed.
```

Comment:
It's true that if you need to customize based on options you need to use the macro but you also need it if you do any custom code generation based on the struct itself.

It's also worth saying that you need to either specify `@fallback_to_any true` or implement `__deriving__` for deriving to work.